low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -20,49 +20,247 @@
+        Service.inactive.discard(service.name)
+
+
+class InactiveState:
+    failed = False
+
+    @staticmethod
+    def enter(service):
+        service.inactive_enter_timestamp = time.time()
+        Service.inactive.add(service.name)
+
+
+class DisabledState:
+
+    @staticmethod
+    def enter(service):
+        logger.warning("{} service is disabled, it will not be "
+                       "monitored".format(service.name))
+        Service.inactive.discard(service.name)
+        Service.monitoring_disabled.discard(service.name)
+        if service.properties_changed_signal:
+            service.properties_changed_signal.remove()
+
+
+class EnabledState:
+
+    @staticmethod
+    def enter(service):
+        try:
+            ServiceMonitor.subscribe_properties_changed_signal(service)
+            Service.monitoring_disabled.discard(service.name)
+            # Call properties_changed_handler to move service into correct state
+            # on start up or if service is enabled after start up
+            service.properties_changed_handler("", "", "")
+        except DBusException:
+            service.new_unit_state(MonitoringDisabled)
+
+
+class MonitoringDisabled:
+
+    @staticmethod
+    def enter(service):
+        Service.alerts.put((service.name, FailedAlert))
+        Service.monitoring_disabled.add(service.name)
+
+
+class Service:
+    """
+    The Service class representing systemd service.
+    """
+
+    alerts = Queue()
+    inactive = set()
+    monitoring_disabled = set()
+
+    def __init__(self, unit):
+        """
+        Initialize Service.
+        """
+        self.unit = unit
+        self.properties_iface = Interface(self.unit,
+                                          dbus_interface=PROPERTIES_IFACE)
+        self.name = str(self.properties_iface.Get(UNIT_IFACE, 'Id'))
+        self.state = "N/A"
+        self.substate = "N/A"
+        self.pid = "N/A"
+        self.previous_state = "N/A"
+        self.previous_substate = "N/A"
+        self.previous_pid = "N/A"
+        self.inactive_enter_timestamp = time.time()
+        self.inactive_threshold = int(
+            Conf.get(SSPL_CONF,
+                     f"{ServiceMonitor.name().upper()}>threshold_inactive_time",
+                     '60'))
+        self.properties_changed_signal = None
+        self._service_state = ActiveState
+        self._unit_state = None
+
+    @property
+    def enabled_on_disk(self):
+        return 'disabled' not in self.properties_iface.Get(
+            UNIT_IFACE, 'UnitFileState')
+
+    @property
+    def failed(self):
+        return self._service_state.failed
+
+    def is_inactive_for_threshold_time(self):
+        return time.time() - self.inactive_enter_timestamp > self.inactive_threshold
+
+    def new_service_state(self, new_state):
+        if self.is_valid_state_change(new_state):
+            self._service_state = new_state
+            new_state.enter(self)
+
+    def is_valid_state_change(self, new_state):
```

Comment:
if service was in deactivating state(or any other non active state) for more than 180 seconds, alert will be raised and service state will be FailedState. Now, if service becomes inactive, we don't want to raise alert. If we Make service's state Inactive then 180 second it will raise alert again(if no further state changes)
So, FailedState to InactiveState is invalid transition.